src/main/kotlin/org/rust/lang/core/types/ty/TyPrimitive.kt

```diff
@@ -56,6 +56,11 @@ object TyUnit : TyPrimitive {
     override fun toString(): String = tyToString(this)
 }
 
+/** The `!` type. E.g. `unimplemented!()` */
+object TyNever : TyPrimitive {
+    override fun toString(): String = tyToString(this)
```

Comment:
`tyToString` should be updated as well. I wonder whether we should make `TyPrimitive` a sealed class... IIRC, it is not possible for some stupid reason, but I may be recalling wrgon!